v2/internal/reconcilers/arm/arm_client_cache.go

```diff
@@ -9,6 +9,7 @@ import (
 	"context"
 	"net/http"
 	"reflect"
+	"strings"
 	"sync"
 
```

Comment:
What other stuff is remaining for this?

1. WorkloadIdentity/ManagedIdentity support
2. E2E tests?

Are you planning on doing those in separate PRs? It would be good to get E2E tests ASAP IMO.

Reply:
and Secret watcher. 

Yep, for e2e tests we still need to discuss how to use this in `testContext`.